### PR TITLE
Store Floats as Strings

### PR DESCRIPTION
**Pull request by theAnnie**
Addresses #438

### CODE REVIEW

**Comment by theAnnie** [comment]
This will fail so far - we are losing rest of the  numer after `.`,  searching for the issue

File: src/format.rs [hunk 1]
```diff
@@ -313,7 +313,7 @@ impl Documentable for &UntypedPattern {
         match self {
             Pattern::Int { value, .. } => value.clone().to_doc(),
 
-            Pattern::Float { value, .. } => value.to_doc(),
+            Pattern::Float { value, .. } => value.clone().to_doc(),
```
Comment (lpil) [inline src/format.rs]: Gleam allows floats to omit the numbers after the `.`. In the Erlang we want the number to always have a number after the `.` so here we can check to see if the `value` string ends with `.` and add a 0 to the end if it does: `format!("{}0", value).to_doc()`

**Comment by lpil** [comment]
Looking good! I've left a comment on what I think the issue is

File: src/format.rs [hunk 2]
```diff
@@ -313,7 +313,13 @@ impl Documentable for &UntypedPattern {
         match self {
             Pattern::Int { value, .. } => value.clone().to_doc(),
 
-            Pattern::Float { value, .. } => value.to_doc(),
+            Pattern::Float { value, .. } => {  
+                if value.ends_with(".") {
+                    format!("{}0", value).to_doc()
+                } else {
+                    value.clone().to_doc()
+                }
```
Comment (theAnnie) [inline src/format.rs]: I'm not sure if thats a good approach. Also for some reason `value.ends_with(".")` always returns false

Comment (lpil) [reply src/format.rs]: Sorry, I have directed you to the wrong file. We want the Erlang printer, not the Gleam formatter.

Comment (theAnnie) [reply src/format.rs]: np, fixed 😸

**Comment by theAnnie** [comment]
> Thank you!

You're welcome 😄

**Pull request merged**